pkg/lockfile/matcher-dependency-map.go

```diff
@@ -118,7 +120,17 @@ func propagateDepGroups(root *PackageDetails, visitedMap map[*PackageDetails]str
 		for _, group := range deps.DepGroups {
 			newDepGroups[group] = true
 		}
-		deps.DepGroups = slices.Collect(maps.Keys(newDepGroups))
+	}
+
+	// Compute the slice once outside the loop instead of N times
+	var mergedGroups []string
+	if len(newDepGroups) > 0 {
+		mergedGroups = slices.Collect(maps.Keys(newDepGroups))
+	}
+
+	// Just assign the same slice to all children (simple assignment, no allocation)
+	for _, deps := range root.Dependencies {
+		deps.DepGroups = mergedGroups
```

Comment:
Each `deps.DepGroups` will reference the same `mergedGroups` slice in this case. Meaning if you add to the slice for one dep, all deps will get that update -- is that desired? Any side-effects?